### PR TITLE
Update TicTacToe Functionality

### PR DESCRIPTION
**Pull request by davidm009**
I made several changes to the TicTacToe code. I began by using React to build the various components and also to update the state of the game. I also added some CSS code to improve the UI of the game and how it looked on the website.

### CODE REVIEW

File: src/App.css [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+}
+
+@keyframes logo-spin {
+  from {
+    transform: rotate(0deg);
+  }
+  to {
+    transform: rotate(360deg);
+  }
+}
```
Comment (purelogiq) [inline src/App.css]: Hey man, this looks like its from the boilerplate Vite project setup, do we still need this?

File: src/App.jsx [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+    [0, 3, 6], // Columns
+    [1, 4, 7],
+    [2, 5, 8],
+    [0, 4, 8], // Diagonals
+    [2, 4, 6],
+  ];
+
+ 
+
+  const checkWinner = (currentBoard) => {
```
Comment (purelogiq) [inline src/App.jsx:26]: This looks good for this PR, no changes needed right now. But in a future PR do you think we could add support for the tied game functionality?

File: src/App.jsx [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+
+  }
+
+
+  function boxClicker(e) {
+    if (winner) {
+      return
+    }
+ 
+    console.log(e.target);
```
Comment (purelogiq) [inline src/App.jsx]: Hey man I think you left this `console.log` in the file, we don't want to keep that around in the production codebase.

File: src/App.jsx [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+    const index = e.target.id; // Getting ID of clicked square
+    const currentPlayer = count % 2 === 0 ? 'X' : 'O';
+
+    const newColors = [...squareColors];
+    newColors[index] = count % 2 === 0 ? 'blue' : 'red';
+    setSquareColors(newColors);
+    
+    console.log(currentPlayer);
```
Comment (purelogiq) [inline src/App.jsx]: Same here

File: src/App.jsx [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+      </div>
+
+      <div className='row'>
+        <div className="square" id="6"><span style={{color: squareColors[6]}}>{board[6]}</span></div>
+        <div className="square" id="7"><span style={{color: squareColors[7]}}>{board[7]}</span></div>
+        <div className="square" id="8"><span style={{color: squareColors[8]}}>{board[8]}</span></div>
+        
+      </div>
+
+      {winner != null && <button onClick={resetGame} className='button'>Reset Game</button>}
```
Comment (purelogiq) [inline src/App.jsx:99]: This is great love how you added this inline!

File: src/App.jsx [hunk 6]
```diff
@@ -0,0 +1,133 @@
...
+  );
+
+};
+
+function App() {
+  
+  
+
+  
+
```
Comment (purelogiq) [inline src/App.jsx]: Looks like there is a lot of extra space here for no reason

**Review by purelogiq — CHANGES_REQUESTED** [review]
Hey looking pretty good, just a few small changes then we should be good to go!

File: src/App.jsx (hunk 2 above)
Comment (davidm009) [reply src/App.jsx:26]: For sure, let's fix it!

**Review by purelogiq — APPROVED** [review]
LGTM (looks good to me!) ✅

**Pull request merged**